### PR TITLE
MGMT-2128: Change log filenames, including for presigned S3 urls

### PR DESCRIPTION
**Pull request by avishayt**
Cluster log file: ClusterName_ClusterID.tar.gz
Host log file: ClusterName_HostRole_HostName.tar.gz

### CODE REVIEW

**Review by YuviGold — APPROVED** [review]
LGTM

File: pkg/s3wrapper/client.go [hunk 1]
```diff
@@ -119,7 +119,7 @@ func (c *S3Client) IsAwsS3() bool {
 	if c.cfg.S3EndpointURL == "" || strings.HasSuffix(c.cfg.S3EndpointURL, awsEndpointSuffix) {
 		return true
 	}
-	return false
+	return true
```
Comment (YuviGold) [inline pkg/s3wrapper/client.go:122]: Was this change intended? This function always returns `true`.

File: internal/bminventory/inventory_test.go [hunk 2]
```diff
@@ -2619,7 +2617,7 @@ var _ = Describe("Upload and Download logs test", func() {
 		r := ioutil.NopCloser(bytes.NewReader([]byte("test")))
 		mockS3Client.EXPECT().Download(ctx, fileName).Return(r, int64(4), nil)
 		generateReply := bm.DownloadHostLogs(ctx, params)
-		downloadFileName := fmt.Sprintf("%s_%s", hostutil.GetHostnameForMsg(&host), filepath.Base(fileName))
+		downloadFileName := fmt.Sprintf("mycluster_master_%s.tar.gz", newHostID.String())
```
Comment (YuviGold) [inline internal/bminventory/inventory_test.go:2620]: Can you add a test to verify also a `bootstrap` scenario?

File: pkg/s3wrapper/client.go (hunk 1 above)
Comment (avishayt) [reply pkg/s3wrapper/client.go:122]: Wow good call

File: internal/bminventory/inventory_test.go (hunk 2 above)
Comment (avishayt) [reply internal/bminventory/inventory_test.go:2620]: Will do

**Comment by gamli75** [comment]
@avishayt 
few remarks:
1. Are you sure that the cluster/ host name doesn't include characters that are illegal for a file name? 
2. looks like all the code is in the inventory - maybe it should be in the cluster/ host APIs

**Comment by avishayt** [comment]
> @avishayt
> few remarks:
> 
> 1. Are you sure that the cluster/ host name doesn't include characters that are illegal for a file name?
> 2. looks like all the code is in the inventory - maybe it should be in the cluster/ host APIs

@gamli75 
1. Yes, I call sanitize.Name which is supposed to remove any illegal characters.
2. I agree that there is too much code in inventory.go, but that's bigger than this change.  All the log stuff should move to a different handler IMO.

File: internal/cluster/cluster.go [hunk 3]
```diff
@@ -497,13 +502,39 @@ func (m *Manager) CreateTarredClusterLogs(ctx context.Context, c *common.Cluster
...
 	}).([]string)
 
+	uuidRegex := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	var tarredFilenames []string
+	var tarredFilename string
+	for _, file := range files {
+		fileNameSplit := strings.Split(file, "/")
+		tarredFilename = file
+		if len(fileNameSplit) > 1 {
+			if uuidRegex.MatchString(fileNameSplit[len(fileNameSplit)-2]) {
```
Comment (tsorya) [inline internal/cluster/cluster.go:512]: i think _, err := uuid.Parse(u) much easier check

File: internal/cluster/cluster.go [hunk 4]
```diff
@@ -497,13 +502,39 @@ func (m *Manager) CreateTarredClusterLogs(ctx context.Context, c *common.Cluster
 		return x != fileName
 	}).([]string)
 
+	uuidRegex := regexp.MustCompile("^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-4[a-fA-F0-9]{3}-[8|9|aA|bB][a-fA-F0-9]{3}-[a-fA-F0-9]{12}$")
+	var tarredFilenames []string
+	var tarredFilename string
+	for _, file := range files {
+		fileNameSplit := strings.Split(file, "/")
+		tarredFilename = file
+		if len(fileNameSplit) > 1 {
```
Comment (tsorya) [inline internal/cluster/cluster.go:511]: maybe better to change it to :
if len(fileNameSplit) < 1 {
continue
}
